app/electron/main.ts

```diff
@@ -37,6 +37,8 @@ const defaultPort = 4466;
 const isDev = process.env.ELECTRON_DEV || false;
 const useExternalServer = process.env.EXTERNAL_SERVER || false;
 const shouldCheckForUpdates = process.env.HEADLAMP_CHECK_FOR_UPDATES !== 'false';
+// make it global so that it doesn't get garbage collected
+let mainWindow: BrowserWindow | null;
```

Comment:
When why would this be gc'd?

Reply:
on app close only

Reply:
I think my question was written in a confusing/wrong way.
Can you elaborate on what was the issue with having the window be declared in the startElectron method? (since it's only used there)

Reply:
It's an extra problem i was looking for which can sometime make the main process block and hence laggy experience, This has to do with local scope defining of mainWindow which can get garbage collected on function execution,

Making it a global scope one prevents us from mistakenly getting the mainWindow varibale getting garbage collected.

Reply:
checkout this 
https://stackoverflow.com/questions/71810712/electron-app-freezes-in-production-build-when-there-are-no-errors-and-no-memory

Reply:
yeah this is more of a defensive solution to prevent any unnecessary gotchas.
https://github.com/electron/electron-quick-start/issues/221